src/Workspaces/Core/Portable/Workspace/Host/TaskScheduler/ITaskSchedulerProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// A factory that creates either sequential or parallel task schedulers.
+    /// </summary>
+    internal interface ITaskSchedulerProvider : IWorkspaceService
+    {
+        TaskScheduler GetCurrentContextScheduler();
```

Comment:
consider having this be a property.  It's unclear to me what computation this would ever perform (but maybe i'm off).

Reply:
https://github.com/dotnet/roslyn/pull/42610/files#diff-43bb72b036bebcb74c466541bdda431eR29 actually constructs a new scheduler. Maybe we can use a singleton there?

Reply:
yeah, singleton seems appropriate here.

Reply:
Using singleton and property.